error[E0277]: the trait bound `valuable::Value<'_>: tracing::Value` is not satisfied

## Describe the bug
  Compiling bytesize v1.3.0
error[E0277]: the trait bound `valuable::Value<'_>: tracing::Value` is not satisfied
  --> /home/ubuntu-mate/.cargo/registry/src/index.crates.io-6f17d22bba15001f/caligula-0.4.7/src/escalated_daemon/mod.rs:54:9
   |
54 |         info!(msg = msg.as_value(), "Received SpawnWriter request");
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `tracing::Value` is not implemented for `valuable::Value<'_>`
   |
   = help: the following other types implement trait `tracing::Value`:
             bool
             isize
             i8
             i16
             i32
             i64
             i128
             usize
           and 34 others
   = note: required for the cast from `&valuable::Value<'_>` to `&dyn tracing::Value`
   = note: this error originates in the macro `$crate::valueset` which comes from the expansion of the macro `info` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `valuable::Value<'_>: tracing::Value` is not satisfied
   --> /home/ubuntu-mate/.cargo/registry/src/index.crates.io-6f17d22bba15001f/caligula-0.4.7/src/ui/herder/herder.rs:109:9
    |
109 | /         debug!(
110 | |             first_msg = first_msg.as_value(),
111 | |             "Read raw result from child"
112 | |         );
    | |_________^ the trait `tracing::Value` is not implemented for `valuable::Value<'_>`
    |
    = help: the following other types implement trait `tracing::Value`:
              bool
              isize
              i8
              i16
              i32
              i64
              i128
              usize
            and 34 others
    = note: required for the cast from `&valuable::Value<'_>` to `&dyn tracing::Value`
    = note: this error originates in the macro `$crate::valueset` which comes from the expansion of the macro `debug` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `valuable::Value<'_>: tracing::Value` is not satisfied
  --> /home/ubuntu-mate/.cargo/registry/src/index.crates.io-6f17d22bba15001f/caligula-0.4.7/src/childproc_common.rs:34:5
   |
34 | /     info!(
35 | |         ?socket_path,
36 | |         args = args.as_value(),
37 | |         "We are in child process mode"
38 | |     );
   | |_____^ the trait `tracing::Value` is not implemented for `valuable::Value<'_>`
   |
   = help: the following other types implement trait `tracing::Value`:
             bool
             isize
             i8
             i16
             i32
             i64
             i128
             usize
           and 34 others
   = note: required for the cast from `&valuable::Value<'_>` to `&dyn tracing::Value`
   = note: this error originates in the macro `$crate::valueset` which comes from the expansion of the macro `info` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `caligula` (bin "caligula") due to 6 previous errors
error: failed to compile `caligula v0.4.7`, intermediate artifacts can be found at `/tmp/cargo-installLtgEm5`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
ubuntu-mate@ubuntu-mate:~$ 

### To Reproduce
Steps to reproduce the behavior:
1. boot to Ubuntu Mate live USB
2. apt-get upgrade && apt-get dist-upgrade && apt-get install cargo
3. cargo install caligula
4. See error

### Expected behavior
A clear and concise description of what you expected to happen.

### Log files
Please upload log files to aid in debugging the issue.

## OS/Environment
 - Terminal emulator: MATE Terminal 1.26.1 powered by Virtual TErminal 0.76.0
 - OS: Linux ubuntu-mate 6.8.0-31-generic #&#8203;31-Ubuntu SMP PREEMPT_DYNAMIC Sat Apr 20 00:40:06 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
 - Version: cargo 1.75.0, caligula v0.4.7, rustc 1.75.0 (82e1608df 2023-12-21) (built from a source tarball)
 - Distribution channel: cargo

## Additional context
ubuntu-mate@ubuntu-mate:~$ ls /tmp/cargo-installLtgEm5/release/deps/libbytesi*
/tmp/cargo-installLtgEm5/release/deps/libbytesize-888f1ee174381d35.rlib
/tmp/cargo-installLtgEm5/release/deps/libbytesize-888f1ee174381d35.rmeta

Well, I thought I had made a reproducible minimal test case, but today I got a different result.  I tried the command line RUST_LOG=debug cargo install caligula, and the error happened after the line for base64 instead of bytesize, but the part of the source code where the tracing::Value error appeared is the same.  No "debug" lines were printed in the output, so I'm not sure whether that was useful.  I'm very new to rust.

   Compiling base64 v0.22.1
error[E0277]: the trait bound `valuable::Value<'_>: tracing::Value` is not satisfied
  --> /home/ubuntu-mate/.cargo/registry/src/index.crates.io-6f17d22bba15001f/caligula-0.4.7/src/escalated_daemon/mod.rs:54:9
   |
54 |         info!(msg = msg.as_value(), "Received SpawnWriter request");
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `tracing::Value` is not implemented for `valuable::Value<'_>`
   |
   = help: the following other types implement trait `tracing::Value`:
             bool
             isize
             i8
             i16
             i32
             i64
             i128
             usize
           and 34 others
   = note: required for the cast from `&valuable::Value<'_>` to `&dyn tracing::Value`
   = note: this error originates in the macro `$crate::valueset` which comes from the expansion of the macro `info` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `valuable::Value<'_>: tracing::Value` is not satisfied
   --> /home/ubuntu-mate/.cargo/registry/src/index.crates.io-6f17d22bba15001f/caligula-0.4.7/src/ui/herder/herder.rs:109:9
    |
109 | /         debug!(
110 | |             first_msg = first_msg.as_value(),
111 | |             "Read raw result from child"
112 | |         );
    | |_________^ the trait `tracing::Value` is not implemented for `valuable::Value<'_>`
    |
    = help: the following other types implement trait `tracing::Value`:
              bool
              isize
              i8
              i16
              i32
              i64
              i128
              usize
            and 34 others
    = note: required for the cast from `&valuable::Value<'_>` to `&dyn tracing::Value`
    = note: this error originates in the macro `$crate::valueset` which comes from the expansion of the macro `debug` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `valuable::Value<'_>: tracing::Value` is not satisfied
  --> /home/ubuntu-mate/.cargo/registry/src/index.crates.io-6f17d22bba15001f/caligula-0.4.7/src/childproc_common.rs:34:5
   |
34 | /     info!(
35 | |         ?socket_path,
36 | |         args = args.as_value(),
37 | |         "We are in child process mode"
38 | |     );
   | |_____^ the trait `tracing::Value` is not implemented for `valuable::Value<'_>`
   |
   = help: the following other types implement trait `tracing::Value`:
             bool
             isize
             i8
             i16
             i32
             i64
             i128
             usize
           and 34 others
   = note: required for the cast from `&valuable::Value<'_>` to `&dyn tracing::Value`
   = note: this error originates in the macro `$crate::valueset` which comes from the expansion of the macro `info` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `caligula` (bin "caligula") due to 6 previous errors
error: failed to compile `caligula v0.4.7`, intermediate artifacts can be found at `/tmp/cargo-installDvXwx8`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
ubuntu-mate@ubuntu-mate:~$ 

I'm not understanding what to do next to solve this problem. Any suggestions welcome. Thank you.

I've now tried this on Ubuntu Jammy as well. It seems that this is happening in the same source files, and the exact compilation step that is showing in the cargo workflow is not very relevant.  I'm changing the title of this to reflect the trait bound error, hoping that this will be closer to the real issue.

   Compiling shell-words v1.1.0
error[E0277]: the trait bound `valuable::Value<'_>: tracing::Value` is not satisfied
  --> /home/bryan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/caligula-0.4.7/src/escalated_daemon/mod.rs:54:9
   |
54 |         info!(msg = msg.as_value(), "Received SpawnWriter request");
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `tracing::Value` is not implemented for `valuable::Value<'_>`
   |
   = help: the following other types implement trait `tracing::Value`:
             bool
             isize
             i8
             i16
             i32
             i64
             i128
             usize
           and 34 others
   = note: required for the cast from `&valuable::Value<'_>` to `&dyn tracing::Value`
   = note: this error originates in the macro `$crate::valueset` which comes from the expansion of the macro `info` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `valuable::Value<'_>: tracing::Value` is not satisfied
   --> /home/bryan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/caligula-0.4.7/src/ui/herder/herder.rs:109:9
    |
109 | /         debug!(
110 | |             first_msg = first_msg.as_value(),
111 | |             "Read raw result from child"
112 | |         );
    | |_________^ the trait `tracing::Value` is not implemented for `valuable::Value<'_>`
    |
    = help: the following other types implement trait `tracing::Value`:
              bool
              isize
              i8
              i16
              i32
              i64
              i128
              usize
            and 34 others
    = note: required for the cast from `&valuable::Value<'_>` to `&dyn tracing::Value`
    = note: this error originates in the macro `$crate::valueset` which comes from the expansion of the macro `debug` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `valuable::Value<'_>: tracing::Value` is not satisfied
  --> /home/bryan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/caligula-0.4.7/src/childproc_common.rs:34:5
   |
34 | /     info!(
35 | |         ?socket_path,
36 | |         args = args.as_value(),
37 | |         "We are in child process mode"
38 | |     );
   | |_____^ the trait `tracing::Value` is not implemented for `valuable::Value<'_>`
   |
   = help: the following other types implement trait `tracing::Value`:
             bool
             isize
             i8
             i16
             i32
             i64
             i128
             usize
           and 34 others
   = note: required for the cast from `&valuable::Value<'_>` to `&dyn tracing::Value`
   = note: this error originates in the macro `$crate::valueset` which comes from the expansion of the macro `info` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `caligula` (bin "caligula") due to 6 previous errors
error: failed to compile `caligula v0.4.7`, intermediate artifacts can be found at `/tmp/cargo-installs2FCuo`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.

--- a/.cargo/config.toml
+++ b/.cargo/config.toml
@@ -1,2 +1,0 @@
-[build]
-rustflags = ["--cfg", "tracing_unstable"]

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -47,11 +47,9 @@ tracing = { version = "0.1.40", features = [
     "async-await",
     "log",
     "release_max_level_debug",
-    "valuable",
 ] }
 tracing-subscriber = { version = "0.3.16", features = ["env-filter", "fmt"] }
 tracing-unwrap = "1.0.1"
-valuable = { version = "0.1.0", features = ["derive"] }
 which = "6.0.1"
 xz2 = { version = "0.1.7", features = ["static"] }
 

--- a/src/childproc_common.rs
+++ b/src/childproc_common.rs
@@ -4,7 +4,6 @@ use serde::de::DeserializeOwned;
 use std::fmt::Debug;
 use tracing::{error, info};
 use tracing_unwrap::ResultExt;
-use valuable::Valuable;
 
 use crate::logging::init_logging_child;
 
@@ -18,7 +17,7 @@ use crate::logging::init_logging_child;
 /// - Get the child-specific config from arg 3
 ///
 /// This returns the socket path and the child-specific config.
-pub fn child_init<C: DeserializeOwned + Debug + Valuable>() -> (String, C) {
+pub fn child_init<C: DeserializeOwned + Debug>() -> (String, C) {
     let cli_args: Vec<String> = env::args().collect();
 
     // We will set up logging first because if any part of this godforsaken
@@ -31,11 +30,7 @@ pub fn child_init<C: DeserializeOwned + Debug + Valuable>() -> (String, C) {
 
     let socket_path = &cli_args[2];
     let args = serde_json::from_str::<C>(&cli_args[3]).unwrap_or_log();
-    info!(
-        ?socket_path,
-        args = args.as_value(),
-        "We are in child process mode"
-    );
+    info!(?socket_path, ?args, "We are in child process mode");
 
     (socket_path.clone(), args)
 }

--- a/src/compression.rs
+++ b/src/compression.rs
@@ -8,7 +8,6 @@ use std::{
 };
 
 use serde::{Deserialize, Serialize};
-use valuable::Valuable;
 
 macro_rules! generate {
     {
@@ -47,7 +46,7 @@ macro_rules! generate {
             }
         }
 
-        #[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize, Valuable)]
+        #[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
         pub enum CompressionFormat {
             Identity,
             $(

--- a/src/device.rs
+++ b/src/device.rs
@@ -7,7 +7,6 @@ use std::{
 
 use bytesize::ByteSize;
 use serde::{Deserialize, Serialize};
-use valuable::Valuable;
 
 #[cfg(target_os = "linux")]
 pub fn enumerate_devices() -> impl Iterator<Item = WriteTarget> {
@@ -307,7 +306,7 @@ impl Display for Removable {
     }
 }
 
-#[derive(Debug, PartialEq, Eq, Clone, Copy, Serialize, Deserialize, Valuable)]
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Serialize, Deserialize)]
 pub enum Type {
     File,
     Disk,

--- a/src/escalated_daemon/ipc.rs
+++ b/src/escalated_daemon/ipc.rs
@@ -1,14 +1,13 @@
 use serde::{Deserialize, Serialize};
-use valuable::Valuable;
 
 use crate::writer_process::ipc::WriterProcessConfig;
 
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Valuable)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub struct EscalatedDaemonInitConfig {
     // Okay, there's nothing here right now, but there might be someday!
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Valuable)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub struct SpawnWriter {
     pub log_file: String,
     pub init_config: WriterProcessConfig,

--- a/src/escalated_daemon/mod.rs
+++ b/src/escalated_daemon/mod.rs
@@ -18,7 +18,6 @@ use interprocess::local_socket::{tokio::prelude::*, GenericFilePath};
 use tokio::io::{AsyncBufRead, BufReader};
 use tracing::{error, info, info_span, Instrument};
 use tracing_unwrap::ResultExt;
-use valuable::Valuable;
 
 use crate::{
     childproc_common::child_init,
@@ -51,7 +50,7 @@ pub async fn main() {
 async fn event_loop(socket: &str, mut stream: impl AsyncBufRead + Unpin) -> anyhow::Result<()> {
     loop {
         let msg = read_msg_async::<SpawnWriter>(&mut stream).await?;
-        info!(msg = msg.as_value(), "Received SpawnWriter request");
+        info!(?msg, "Received SpawnWriter request");
 
         let command =
             make_writer_spawn_command(socket.into(), msg.log_file.into(), &msg.init_config);

--- a/src/hash.rs
+++ b/src/hash.rs
@@ -3,7 +3,6 @@ use digest::Digest;
 use serde::{Deserialize, Serialize};
 use std::fmt::Display;
 use std::io::Read;
-use valuable::Valuable;
 
 macro_rules! generate {
     {$(
@@ -17,7 +16,7 @@ macro_rules! generate {
             })*
         ]
     )*} => {
-        #[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize, Valuable)]
+        #[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
         pub enum HashAlg {
             $($(
                 $enum_arm,

--- a/src/run_mode.rs
+++ b/src/run_mode.rs
@@ -2,7 +2,6 @@ use std::{borrow::Cow, fmt::Debug};
 
 use process_path::get_executable_path;
 use serde::Serialize;
-use valuable::Valuable;
 
 use crate::{
     escalated_daemon::ipc::EscalatedDaemonInitConfig, escalation::Command,
@@ -67,7 +66,7 @@ impl RunMode {
 }
 
 /// Build a [Command] that, when run, spawns a process with a specific configuration.
-pub fn make_spawn_command<'a, C: Serialize + Debug + Valuable>(
+pub fn make_spawn_command<'a, C: Serialize + Debug>(
     socket: Cow<'a, str>,
     log_path: Cow<'a, str>,
     run_mode: RunMode,

--- a/src/ui/herder/herder.rs
+++ b/src/ui/herder/herder.rs
@@ -12,7 +12,6 @@ use crate::{
 use anyhow::Context;
 use interprocess::local_socket::tokio::prelude::*;
 use tracing::{debug, trace};
-use valuable::Valuable;
 
 use crate::escalation::run_escalate;
 use crate::writer_process::ipc::{StatusMessage, WriterProcessConfig};
@@ -109,10 +108,7 @@ impl Herder {
 
         trace!("Reading results from child");
         let first_msg = read_msg_async::<StatusMessage>(&mut handle.rx).await?;
-        debug!(
-            first_msg = first_msg.as_value(),
-            "Read raw result from child"
-        );
+        debug!(?first_msg, "Read raw result from child");
 
         let initial_info = match first_msg {
             StatusMessage::InitSuccess(i) => Ok(i),

--- a/src/writer_process/ipc.rs
+++ b/src/writer_process/ipc.rs
@@ -2,12 +2,10 @@ use std::{fmt::Display, path::PathBuf};
 
 use serde::{Deserialize, Serialize};
 
-use valuable::Valuable;
-
 use crate::compression::CompressionFormat;
 use crate::device::Type;
 
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Valuable)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub struct WriterProcessConfig {
     pub dest: PathBuf,
     pub src: PathBuf,
@@ -17,7 +15,7 @@ pub struct WriterProcessConfig {
     pub block_size: Option<u64>,
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Valuable)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub enum StatusMessage {
     InitSuccess(InitialInfo),
     TotalBytes {
@@ -37,12 +35,12 @@ pub enum StatusMessage {
     Error(ErrorType),
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Valuable)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub struct InitialInfo {
     pub input_file_bytes: u64,
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Valuable)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub enum ErrorType {
     EndOfOutput,
     PermissionDenied,

